XenAdmin/Dialogs/OptionsPages/ConnectionOptionsPage.cs

```diff
@@ -170,11 +228,31 @@ public void Save()
             if (ProxyAddressTextBox.Text != Properties.Settings.Default.ProxyAddress && ProxyAddressTextBox.Text != "")
                 Properties.Settings.Default.ProxyAddress = ProxyAddressTextBox.Text;
 
-            if (new_proxy_style == HTTPHelper.ProxyStyle.SpecifiedProxy)
+            Properties.Settings.Default.ProxyUsername = EncryptionUtils.Protect(ProxyUsernameTextBox.Text);
+            Properties.Settings.Default.ProxyPassword = EncryptionUtils.Protect(ProxyPasswordTextBox.Text);
+
+            if (int.Parse(ProxyPortTextBox.Text) != Properties.Settings.Default.ProxyPort)

```

Comment:
No exception handling in line 234. Also, consider refactoring to avoid parsing twice.